backend/app/services/accounts/with_email_notification.rb

```diff
@@ -0,0 +1,28 @@
+module Accounts
+  class WithEmailNotification
```

Comment:
Are there more methods than `destroy!` planned for this service? I find it little weird to be about email notifications than destroying, have you thought about services like `Accounts::Destroy`, than would just have `call` or `perform` method?

Reply:
I imagine when adding more services there could be more actions performed than just sending email notification on common actions like `destroy` or `update`

Reply:
I could not actually decide :). My first idea was to write it as wrapper (use `method_missing` to forward all stuff to main object), but then I got feeling that maybe it is more robust approach then it is required now. So in the end, wrapper looks more like service class and from this view, `call` or `perform` makes more sense. What do you think? Should I finalise wrapper idea and add `method_missing` stuff which can be later on enhanced with email notification on update/create/etc. when it gets too complex at controller or should I transform it to single oriented service with `call` method?

Reply:
I'm not sure what do you mean by wrapper? I just thought about changing names of services to be name of actual command so here `Accounts::Destroy`, the inside of the service looks perfectly fine :)

On the other hand service objects could quickly get messy without good conventions, and as there are so many conventions out there so I'm also not sure what would be the best.

Reply:
I think I get what you mean Tomasz, i.e. service called `Accounts::Destroy` and `destroy!` method renamed to `call` - I think I prefer it this way

Reply:
Okay, I still like idea of wrapper/decorator, but truth is that it is over-engineering at this point so I give up on this approach for now :D. My idea was that we could do something similar at future `3rdPartyNotification(SmsNotification(EmailNotification(account))).destroy!` or `SmsNotification(EmailNotification(account)).update!(params)` which makes it easy to combine and encapsulate all notification responsibilities.

Whatever, it is overkill now, will rename services to `Accounts::Destroy` so it follows our services approach ;)